chain-impl-mockchain/benches/tally.rs

```diff
@@ -0,0 +1,193 @@
+            .committee_keys(committee_keys)
+            .with_proposal(
+                proposal(VoteTestGen::external_proposal_id())
+                    .options(3)
+                    .action_transfer_to_rewards(100),
+            )])
+        .build()
+        .unwrap();
+
+    let mut alice = controller.wallet(ALICE).unwrap();
+
+    let vote_plan_def = controller.vote_plan(VOTE_PLAN).unwrap();
+    let vote_plan: VotePlan = vote_plan_def.clone().into();
+    let proposal = vote_plan_def.proposal(0);
+
+    for alias in voters_aliases {
+        let mut private_voter = controller.wallet(&alias).unwrap();
+
+        controller
+            .cast_vote_private(
+                &private_voter,
+                &vote_plan_def,
+                &proposal.id(),
+                favorable,
+                &mut ledger,
+                &mut rng,
+            )
+            .unwrap();
+        private_voter.confirm_transaction();
+    }
+
+    ledger.fast_forward_to(BlockDate {
+        epoch: 1,
+        slot_id: 1,
+    });
+
+    let encrypted_tally = EncryptedVoteTally::new(vote_plan.to_id());
+    let fragment = controller
+        .fragment_factory()
+        .vote_encrypted_tally(&alice, encrypted_tally);
+
+    let parameters = ledger.parameters.clone();
+    let date = ledger.date();
+
+    let bench_name_suffix = format!("_{}_voters_{}_ada", voters_count, voting_power_per_voter);
+
+    c.bench_function(&format!("vote_encrypted_tally{}", bench_name_suffix), |b| {
+        b.iter(|| {
+            ledger
+                .ledger
+                .apply_fragment(&parameters, &fragment, date)
+                .unwrap();
+        })
+    });
+
+    ledger.apply_fragment(&fragment, ledger.date()).unwrap();
+    alice.confirm_transaction();
+
+    let vote_plans = ledger.ledger.active_vote_plans();
+    let vote_plan_status = vote_plans
+        .iter()
+        .find(|c_vote_plan| {
+            let vote_plan: VotePlan = vote_plan.clone().into();
+            c_vote_plan.id == vote_plan.to_id()
+        })
+        .unwrap();
+
+    c.bench_function(&format!("decrypt_tally{}", bench_name_suffix), |b| {
+        b.iter(|| {
+            decrypt_tally(vote_plan_status, &members);
+        })
+    });
+
+    let shares = decrypt_tally(vote_plan_status, &members);
+
+    let decrypted_tally = VoteTally::new_private(vote_plan.to_id(), shares);
+    let fragment = controller
+        .fragment_factory()
+        .vote_tally(&alice, decrypted_tally);
+
+    c.bench_function(&format!("vote_tally{}", bench_name_suffix), |b| {
+        b.iter(|| {
+            ledger
+                .ledger
+                .apply_fragment(&parameters, &fragment, ledger.date())
+                .unwrap();
+        })
+    });
+
+    ledger.apply_fragment(&fragment, ledger.date()).unwrap();
+}
+
+fn tally_benchmark_128_voters_1000_ada(c: &mut Criterion) {
+    tally_benchmark(128, 1000, c);
+}
+
+fn tally_benchmark_200_voters_1000_ada(c: &mut Criterion) {
```

Comment:
This seems quite similar to the test case above it

Reply:
The point was to test with different total voting power. For the previous implementation the voting power made a huge difference.

Reply:
What I mean is that the number of voters does not influence the decryption speed. The advantage of having two benchmarks with similar values in both voters and total voting power (`128_voters_1000_ada` and `200_voters_1000_ada`) is not much IMHO if we already have  bigger tests like `200_votes_1_000_000_ada` or `1000_voters_1000_ada`, but maybe I am missing something. 

Anyway, this is just a nit.

Reply:
> What I mean is that the number of voters does not influence the decryption speed

But it does influence the speed of construction of the encrypted tally.